backend/src/infra/dto/ReimbursementPlan.dto.ts

```diff
@@ -0,0 +1,61 @@
+import type { ReimbursementPlan } from "src/core/entity/ReimbursementPlan";
+
+export class ReimbursementPlanDTO {
+  groupId: number;
+  membersReimbursements: MemberReimbursements[];
+
+  constructor(groupId: number, membersReimbursements: MemberReimbursements[]) {
+    this.groupId = groupId;
+    this.membersReimbursements = membersReimbursements;
+  }
+
+  public static fromEntity(
+    reimbursementPlan: ReimbursementPlan
+  ): ReimbursementPlanDTO {
+    const newMemberReimbursements: MemberReimbursements[] = [];
+    reimbursementPlan.reimbursementPerMemberId.forEach(
+      (memberReimbursementsMap, memberId) => {
+        newMemberReimbursements.push(
+          MemberReimbursements.fromMap(memberId, memberReimbursementsMap)
+        );
+      }
+    );
+    return new ReimbursementPlanDTO(
+      reimbursementPlan.groupId,
+      newMemberReimbursements
+    );
+  }
+}
+
+export class MemberReimbursements {
+  memberId: number;
+  memberDueSums: MemberDueSumsToBeneficiary[];
+
+  constructor(memberId: number, memberDueSums: MemberDueSumsToBeneficiary[]) {
+    this.memberId = memberId;
+    this.memberDueSums = memberDueSums;
+  }
+
+  public static fromMap(
+    memberId: number,
+    memberReimbursementsMap: Map<number, number>
+  ): MemberReimbursements {
+    const newMemberDueSums: MemberDueSumsToBeneficiary[] = [];
+    memberReimbursementsMap.forEach((amount, beneficiaryId) => {
+      newMemberDueSums.push(
+        new MemberDueSumsToBeneficiary(beneficiaryId, amount)
+      );
+    });
+    return new MemberReimbursements(memberId, newMemberDueSums);
+  }
+}
+
+export class MemberDueSumsToBeneficiary {
```

Comment:
Since this class won't have any logic, I would have used an interface direclty instead of a class.
It is less code to write and less overhead, but I don't think this has a huge impact in the end.

---

However, I would rename it `DueSums` or `Debt` because `Member` and `ToBeneficiary` don't add any infnormation. And it made it easier to read